### ISSUE TITLE
unavailable `repo.scala-sbt.org` (April 7, 2023)

### ISSUE THREAD

**Issue by xuwei-k** [sbt/sbt#7202]
## steps

```
$ curl https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.dwijnand/sbt-dynver/scala_2.12/sbt_1.0/4.1.1/jars/sbt-dynver.jar
```

or

`plugins.sbt`

```scala
addSbtPlugin("com.dwijnand" % "sbt-dynver" % "4.1.1")
```

## problem

redirect to https://landing.jfrog.com/reactivate-server/scala 😭 

```html
<html>
<head><title>302 Found</title></head>
<body>
<center><h1>302 Found</h1></center>
<hr><center>nginx/1.19.6</center>
</body>
</html>
```

<img width="666" alt="jfrog" src="https://user-images.githubusercontent.com/389787/230516801-25ea1c49-0d23-422f-b1db-c59397104f10.png">

**Comment by eed3si9n** [comment]
We're looking into it. Thanks for your patience.

**Comment by WayneWang12** [comment]
> We're looking into it. Thanks for your patience.

Any update?

**Comment by gervaisb** [comment]
I have a similar issue with https://repo1.maven.org/maven2/com/typesafe/sbt/
<img width="1050" alt="Screenshot 2023-04-07 at 10 29 47" src="https://user-images.githubusercontent.com/1385954/230574789-69c661c0-dea7-4a0e-b98d-d52784219a3a.png">
<img width="668" alt="Screenshot 2023-04-07 at 10 34 59" src="https://user-images.githubusercontent.com/1385954/230574823-1d210fab-0e5e-4fa2-82e7-ab2e6f415bfc.png">

**Comment by eliashourany-incrowd** [comment]
I also have  the same issue

**Comment by fedotxxl** [comment]
stuck with  critical bug release....

**Comment by charlescd** [comment]
It seems to be related to maven central current status : https://status.maven.org/

**Comment by WayneWang12** [comment]
> It seems to be related to maven central current status : https://status.maven.org/

I don't think so. scala-repo is on jfrog's server. And I believe that JFrog promised to support scala for a long time. I just don't know the long is one year(or two?)...

**Comment by 00subra8** [comment]
I had inadvertently deleted my cache and am now stuck without being able to build my application. Any update on this will be highly appreciated pls...

Thank you

**Comment by SethTisue** [comment]
# PLEASE READ

**You can all safely assume there will be an update here once anything changes.**

Please stop asking "any update?" — you are clogging inboxes with pointless noise. Everyone already understands this is a big deal. You're welcome to comment here if you have helpful new information and/or workarounds to share, but otherwise you will simply need to be patient.

@WayneWang12 is correct that this has absolutely nothing to do with the Sonatype outage. They are two entirely separate outages that just happen to be occurring at the same time. As far as we know, the Sonatype outage only affects publishing; Maven Central has continued to serve up existing artifacts just fine. (If it appears otherwise to you, see below at https://github.com/sbt/sbt/issues/7202#issuecomment-1500343476.)

**The problem on this ticket is confined to repo.scala-sbt.org.**

It is unfortunate that JFrog is serving up 302 response codes from that domain, instead of 404 or some such. This can lead to cache corruption locally, either on local developer machines or on a privately hosted Artifactory (or any similar software that caches artifacts).

For the past few years, repo.scala-sbt.org has generously been hosted by JFrog for free through their OSS program. For reasons unknown, it suddenly stopped working. Eugene has contacted JFrog to try to find out what happened.

**Disclaimer:**

It is currently the middle of the night in New Jersey, where Eugene lives, and that's part of why I'm stepping in here with some information. I am not representing any company or organization today; I'm just a guy who tries to help out in this repo, and this week I happen to be a timezone where the sun is currently up.

If you want to come chat and/or commiserate in a more informal environment, there's always the #sbt channel in the [Scala Discord](https://discord.com/invite/scala).

P.S. I assume you all have seen https://xkcd.com/2347/

**Comment by SethTisue** [comment]
## Workaround for some plugins: try a newer version?

Note that repo.scala-sbt.org hasn't accepted new plugins for several years now. It only has old plugins. So for some plugins, you may be able to resolve the issue immediately by upgrading to a newer version of the plugin, if that newer version is available from Maven Central.

For any plugin whose latest version is still only on the old repo, it would make sense to open a ticket such as https://github.com/sbt/sbt-dynver/issues/239 in the hopes of getting a new version published to Maven Central.

That doesn't help poor benighted people stuck on crufty versions of things, but it's progress regardless towards the long term goal of making the leading edge of the plugin ecosystem more robustly available.

(However, I think repo.scala-sbt.org also serves other things besides plugins...? I'm unclear on that.)

**Comment by gervaisb** [comment]
> Maven Central has continued to serve up existing artifacts just fine

@SethTisue since my comment https://github.com/sbt/sbt/issues/7202#issuecomment-1500075325 is off-topic, do you suggest creating another issue for those that cannot download dependencies from https://repo1.maven.org/maven2/com/typesafe/sbt/ ?

I understand it can be another issue. However, it is an issue too.

**Comment by SethTisue** [comment]
@gervaisb perhaps I was hasty. if you are reasonably confident the problem is at the sbt end, yes, please open an issue, describing what you think the bug in sbt is, including supporting evidence

if you're unsure what's going on or where the root cause is and you want to discuss, there's always https://github.com/sbt/sbt/discussions (as well as Discord, though chat rooms are more hit and miss, though a "hit" can come faster, if one comes at all)

**Comment by marcinaylien** [comment]
My issue is that I miss many dependencies from https://repo1.maven.org/maven2/com/lightbend/sbt/ and https://repo1.maven.org/maven2/com/typesafe/sbt/, which seems to be a yet another strange coincidence. I support @gervaisb.

**Comment by SethTisue** [comment]
Maybe sbt is only even looking in those places at all, because it failed to find something that it should have found on repo.scala-sbt.org? I'm just guessing. Please open a separate issue or discussion.

We need to keep this ticket uncluttered so people won't miss the most important information here.

If the separate discussion reaches a useful conclusion, we can record that conclusion here.

**Comment by gervaisb** [comment]
> if you're unsure what's going on or where the root cause is and you want to discuss, there's always https://github.com/sbt/sbt/discussions

For the record, here is the discussion on this issue: https://github.com/sbt/sbt/discussions/7205

Now I'll stop polluting this other issue ;-)

**Comment by SethTisue** [comment]
## Emergency workaround guidance

If you are really desperate to get a build running while you wait for this to be fixed, the following information may be helpful:

[What can I do in an emergency if a plugin I need isn't on the Internet anymore, temporarily or permanently?](https://github.com/sbt/sbt/discussions/7206)

It's just rough notes but maybe people can use it as a gathering point to help each other and share knowhow.

**Comment by SethTisue** [comment]
## Missing sbt dependencies from repo1.maven.org?

If you are seeing errors about missing JARs from `repo1.maven.org/maven2/com/typesafe/sbt` and similar URLs on `repo1.maven.org`,

Our working hypothesis is that this isn't actually a separate problem, but just a consequence of the repo.scala-sbt.org issue. If you doubt that or think you have further insight, there is a separate discussion on this topic at #7205.

**Comment by zuchos-dd** [comment]
I assume that the fact the RPMs to install `sbt` on Red Hat are missing is covered by this ticket as well

```
curl https://repo.scala-sbt.org/scalasbt/rpm/repodata/repomd.xml
<html>
<head><title>302 Found</title></head>
<body>
<center><h1>302 Found</h1></center>
<hr><center>nginx/1.19.6</center>
</body>
</html>
```

**Comment by SethTisue** [comment]
@zuchos-dd yes. I'll update the issue title

**Comment by SethTisue** [comment]
It seems that repo.scala-sbt.org is now back online, but the CDN might still have stale data so everything might not be working again right away yet.

**Comment by eed3si9n** [comment]
Thanks again to [JFrog](https://jfrog.com/) for their continued sponsorship of Scala community.

I'd say give it an hour or so for traffic to settle down and JARs to get copied over to the right servers etc.

**Comment by guizmaii** [comment]
@eed3si9n Can we imagine the Scala Center hosting a mirror of this repository to avoid this situation from happening again maybe?

**Comment by mkurz** [comment]
@eed3si9n Is there a backup somewhere of that JFrog repo - I mean the actual files? I mean in case JFrog won't sponsor anymore or goes out of business, many people would be in huge trouble.
I think this incident was a wake up call to finally get all projects published to maven central. But that doesn't help users that use old versions of e.g. Play Framework that still depend on older artifacts that only exists on the JFrog instance.

**Comment by mkurz** [comment]
Or would it be somehow possible to migrate all the artifacts to maven central? Publishing to Maven central is more or less just uploading files. I know they use a different layout (at least I think so), so I could imagine there could be a way to write like a script that renames the artifact names, put it in the right folders, comes up with the needed checksum files, etc. and then publish it to maven central?

**Comment by eed3si9n** [comment]
We can discuss these options with Scala Center. Taking backups at least would be a good idea. At someone I did have a copy of a backup from Bintray, but since then I've probably deleted that after I pushed the artifacts into Artifactory.

**Comment by mkurz** [comment]
> Taking backups at least would be a good idea.

Holy moly. I think this is overdue. If something gets wrong so many people will be in deep :hankey:.

**Comment by mkurz** [comment]
...I am saying that because like a a year ago I tried to get a Play 2.2 project working again and I was not able to retrieve some necessary dependencies anymore because they were and still are not hosted anywhere anymore. So that was game over.

**Comment by joelhaasnoot** [comment]
It looks like there is still a setting misconfigured 
```sh
curl https://scala.jfrog.io/artifactory/sbt-plugin-releases/com.typesafe.sbt/sbt-native-packager/scala_2.12/sbt_1.0/1.5.2/jars/sbt-native-packager.jar
{
  "errors" : [ {
    "status" : 500,
    "message" : "Could not process download request: Artifactory is not configured with CLOUD_FRONT."
  } ]
}%     
```

**Comment by eed3si9n** [comment]
I've written this proposal to Scala Center - https://github.com/scalacenter/advisoryboard/pull/120

**Comment by adrianlyjak** [comment]
Not sure if this is the right issue at this point, but what's the relationship between repo.scala-sbt.org and com.typesafe.sbt? Do we know if jfrog re-enabled all of the scala repos?

I tried throwing up a play-framework starter project, and am seeing lots of 500s from jfrog hosted orgs: org.foundweekends.giter8 and com.typesafe.sbt like @joelhaasnoot above

**Comment by fliu-123** [comment]
We saw 500 for these artifacts. Looks to be sbt plugin specific. Just posted for visibility.

[error] https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.typesafe.sbt/sbt-native-packager/scala_2.12/sbt_1.0/1.3.22/docs/sbt-native-packager-javadoc.jar: download error: Caught java.io.IOException (Server returned HTTP response code: 500 for URL: https://scala.jfrog.io/artifactory/sbt-plugin-releases/com.typesafe.sbt/sbt-native-packager/scala_2.12/sbt_1.0/1.3.22/docs/sbt-native-packager-javadoc.jar) while downloading https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.typesafe.sbt/sbt-native-packager/scala_2.12/sbt_1.0/1.3.22/docs/sbt-native-packager-javadoc.jar
[error] https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.lightbend/sbt-google-cloud-storage/scala_2.12/sbt_1.0/0.0.10/docs/sbt-google-cloud-storage-javadoc.jar: download error: Caught java.io.IOException (Server returned HTTP response code: 500 for URL: https://scala.jfrog.io/artifactory/sbt-plugin-releases/com.lightbend/sbt-google-cloud-storage/scala_2.12/sbt_1.0/0.0.10/docs/sbt-google-cloud-storage-javadoc.jar) while downloading https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.lightbend/sbt-google-cloud-storage/scala_2.12/sbt_1.0/0.0.10/docs/sbt-google-cloud-storage-javadoc.jar
[error] https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.typesafe.sbt/sbt-native-packager/scala_2.12/sbt_1.0/1.3.22/jars/sbt-native-packager.jar: download error: Caught java.io.IOException (Server returned HTTP response code: 500 for URL: https://scala.jfrog.io/artifactory/sbt-plugin-releases/com.typesafe.sbt/sbt-native-packager/scala_2.12/sbt_1.0/1.3.22/jars/sbt-native-packager.jar) while downloading https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.typesafe.sbt/sbt-native-packager/scala_2.12/sbt_1.0/1.3.22/jars/sbt-native-packager.jar

**Comment by eed3si9n** [comment]
> Not sure if this is the right issue at this point, but what's the relationship between repo.scala-sbt.org and com.typesafe.sbt? Do we know if jfrog re-enabled all of the scala repos?

"repo.scala-sbt.org" is the URL for the sbt Community repo, which hosts old sbt plugins and Linux installers. It's a redirect Nginx instance that redirects to an underlying repository, which has changed from Artifactory Online -> Bintray -> sponsored Artifactory instance. It should be fully re-enabled as of my comment an hour ago.

"com.typesafe.sbt" is an organization name ("group id") used by a few plugins that was originally authored by Typesafe aka Lightbend but often since then maintained by community members, like in the case of sbt-native-packager. I don't know why that one specifically is causing HTTP 500 response.

```
$ curl -L https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.typesafe.sbt/sbt-native-packager/scala_2.12/sbt_1.0/1.3.22/jars/sbt-native-packager.jar
{
  "errors" : [ {
    "status" : 500,
    "message" : "Could not process download request: Artifactory is not configured with CLOUD_FRONT."
  } ]
}
```

I think it's more to do with some CDN configuration internal to Artifactory?

**Comment by eed3si9n** [comment]
My recommendation for sbt-native-packager specifically is to migrate to the newest (or oldest if that makes things easier) version published on Maven Central - https://repo1.maven.org/maven2/com/github/sbt/sbt-native-packager_2.12_1.0/1.9.16/

**Comment by barryoneill** [comment]
In case it's helpful, I'm seeing the same for sbt-compat:

```
[error] https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.dwijnand/sbt-compat/scala_2.12/sbt_1.0/1.2.6/docs/sbt-compat-javadoc.jar: download error: Caught java.io.IOException (Server returned HTTP response code: 500 for URL: https://scala.jfrog.io/artifactory/sbt-plugin-releases/com.dwijnand/sbt-compat/scala_2.12/sbt_1.0/1.2.6/docs/sbt-compat-javadoc.jar) while downloading https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.dwijnand/sbt-compat/scala_2.12/sbt_1.0/1.2.6/docs/sbt-compat-javadoc.jar
```

**Comment by eed3si9n** [comment]
ok. Let me turn off CDN and turn it back on to see what happens.

```
$ curl -L https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.typesafe.sbt/sbt-native-packager/scala_2.12/sbt_1.0/1.3.22/jars/sbt-native-packager.jar > /tmp/foo.jar
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   145  100   145    0     0    939      0 --:--:-- --:--:-- --:--:--   966
100   148    0   148    0     0    268      0 --:--:-- --:--:-- --:--:--   268
```

it's working now?

**Comment by myazinn** [comment]
@eed3si9n nope. Neither 
```
curl -L https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.typesafe.sbt/sbt-native-packager/scala_2.12/sbt_1.0/1.3.22/jars/sbt-native-packager.jar
```
Nor
```
curl -L https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.portable-scala/sbt-platform-deps/scala_2.12/sbt_1.0/1.0.0/docs/sbt-platform-deps-javadoc.jar
```
work for me because of the same error
```
{
  "errors" : [ {
    "status" : 500,
    "message" : "Could not process download request: Artifactory is not configured with CLOUD_FRONT."
  } ]
}
```

**Comment by joel-apollo** [comment]
> it's working now?

Same thing for me

**Comment by ZinoviiZ** [comment]
The errors I personally had initially got away, but I've started experiencing another package resolution error:
> not found: https://repo1.maven.org/maven2/com/typesafe/playq/sbt-plugin_2.12_1.0/2.8.19/sbt-plugin-2.8.19.pom

**Comment by barryoneill** [comment]
for `sbt-compat-javadoc.jar` I'm afraid not, same error:

```
[error] (update) lmcoursier.internal.shaded.coursier.error.FetchError$DownloadingArtifacts: Error fetching artifacts:
[error] https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.dwijnand/sbt-compat/scala_2.12/sbt_1.0/1.2.6/docs/sbt-compat-javadoc.jar: download error: Caught java.io.IOException (Server returned HTTP response code: 500 for URL: https://scala.jfrog.io/artifactory/sbt-plugin-releases/com.dwijnand/sbt-compat/scala_2.12/sbt_1.0/1.2.6/docs/sbt-compat-javadoc.jar) while downloading https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.dwijnand/sbt-compat/scala_2.12/sbt_1.0/1.2.6/docs/sbt-compat-javadoc.jar
```

Doing `curl -L` on either https://scala.jfrog.io/artifactory/sbt-plugin-releases/com.dwijnand/sbt-compat/scala_2.12/sbt_1.0/1.2.6/docs/sbt-compat-javadoc.jar or https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/com.dwijnand/sbt-compat/scala_2.12/sbt_1.0/1.2.6/docs/sbt-compat-javadoc.jar results in the 500 error posted above.

**Comment by eed3si9n** [comment]
ok. Let me turn off CDN for now. And turn it back on over the weekend.

**Comment by barryoneill** [comment]
Confirmed, my build is alive again.  Thank you for your efforts!

**Comment by eed3si9n** [comment]
My guess is that maybe due to the nature of CDN, it takes a while for the contents bits and bucket to replicate to the AWS CloudFront etc, especially for non-US instances.

**Comment by mkurz** [comment]
> The errors I personally had initially got away, but I've started experiencing another package resolution error:
> 
> > not found: https://repo1.maven.org/maven2/com/typesafe/playq/sbt-plugin_2.12_1.0/2.8.19/sbt-plugin-2.8.19.pom

You have a typo: playq should be just play

**Comment by PromanSEW** [comment]
@ZinoviiZ you have typo: play**q**

**Comment by ZinoviiZ** [comment]
> > The errors I personally had initially got away, but I've started experiencing another package resolution error:
> > > not found: https://repo1.maven.org/maven2/com/typesafe/playq/sbt-plugin_2.12_1.0/2.8.19/sbt-plugin-2.8.19.pom
> 
> You have a typo: playq should be just play
@PromanSEW @mkurz 
Oh, you're right, I've accidentaly added it when tried to debug. But now it still failing with multiple previous errors, like: [error]   not found: https://repo1.maven.org/maven2/com/typesafe/sbt/sbt-js-engine_2.12_1.0/1.2.3/sbt-js-engine-1.2.3.pom

**Comment by eed3si9n** [comment]
I am closing this for now.

**Comment by cbusby-revinate** [comment]
I'm still hitting the same issue trying to pull the sbt-launcher, sadly. 

```
curl -L https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/sbt-launch/0.13.16/sbt-launch.jar
{
  "errors" : [ {
    "status" : 500,
    "message" : "Could not process download request: Artifactory is not configured with CLOUD_FRONT."
  } ]
}
```

It doesn't help that jfrog just went into maintenance.

**Comment by mkurz** [comment]
I took a look at the `~/.cache/coursier/v1/https/repo.scala-sbt.org` folder of my machine to find out which libraries on my machine are currently hosted only on repo.scala-sbt.org and would therefore be affected by a total outage. This is what I came up with:
* https://github.com/sbt/sbt-dynver/issues/239
* https://github.com/sbt/sbt-license-report/issues/51
* https://github.com/sbt/sbt-sdlc/issues/3
* https://github.com/sbt/sbt-multi-jvm/issues/59
* https://github.com/sbt/sbt-osgi/issues/77
* https://github.com/cb372/sbt-explicit-dependencies/issues/102
* https://github.com/johanandren/sbt-akka-version-check/issues/1
* https://github.com/sbt/sbt-javaagent/issues/24
* https://github.com/spray/sbt-revolver/issues/112
* https://github.com/lightbend/sbt-publish-rsync/issues/6
* https://github.com/sbt/sbt-boilerplate/issues/32
* https://github.com/jamesward/play-auto-refresh/issues/47

**Comment by gervaisb** [comment]
@eed3si9n , @SethTisue thanks for your work, your time and your quick solution. 👌

**Comment by eed3si9n** [comment]
@cbusby-revinate Thanks for the report. I just turned off CDN for ivy-releases repo.

**Comment by cbusby-revinate** [comment]
> @cbusby-revinate Thanks for the report. I just turned off CDN for ivy-releases repo.

We're actively testing the build step now. Will report back our build status. Thank you for your prompt support @eed3si9n

**Comment by cbusby-revinate** [comment]
@eed3si9n can confirm, our team has been unblocked. Huge thanks to all maintainers of this project. Thanks again!

**Comment by eed3si9n** [comment]
For the record, I just re-enabled CDN download for sbt-plugin-releases. It might take some time for the content to propagate, but this should make access to JARs faster.

**Comment by eed3si9n** [comment]
Actually given the error message

```
{
  "errors" : [ {
    "status" : 500,
    "message" : "Could not process download request: Artifactory is not configured with CLOUD_FRONT."
  } ]
}
```

maybe we no longer have access to CDN.

**Comment by jrudolph** [comment]
> We can discuss these options with Scala Center. Taking backups at least would be a good idea. At someone I did have a copy of a backup from Bintray, but since then I've probably deleted that after I pushed the artifacts into Artifactory.

I've still got a backup from bintray from April 2021 lying around, let me know if you need it.

**Comment by mkurz** [comment]
@jrudolph Probably a good idea to not delete that backup anytime soon :wink:

**Comment by SethTisue** [comment]
> This can lead to cache corruption locally

for a sample command clearing local cache, see https://github.com/sbt/sbt/issues/7211#issuecomment-1505006500

Coursier's cache location is different on different OSes though, as per https://get-coursier.io/docs/cache

**Comment by SethTisue** [comment]
the Scala Center's blog post about the incident (dated April 20, 2023): https://scala-lang.org/blog/2023/04/20/sbt-plugins-community-repository.html

and an update from a couple days ago with some plans: https://contributors.scala-lang.org/t/roadmap-for-the-sbt-community-repository/6195

**Comment by SethTisue** [comment]
for the rest of this story, read https://www.scala-lang.org/blog/2023/04/20/sbt-plugins-community-repository.html

**Comment by SethTisue** [comment]
and for the 2025 chapter of the story, https://github.com/sbt/sbt/issues/8093
